### ISSUE TITLE
W1b: `Boxes3D.to_tensor()` defaults to `'xyzxyz'` and ignores the stored mode, while `Boxes.to_tensor()` defaults to the stored mode

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4251]
`Boxes.to_tensor(mode=None)` exports in the mode the object was built with (documented on the class since #4060: the label is "reused as the default output mode by `to_tensor`"). `Boxes3D.to_tensor(mode="xyzxyz")` has a hard default and never reads `self._mode`, so `Boxes3D.from_tensor(x, "xyzwhd").to_tensor()` does not round-trip where the 2D twin does. The label is still stored and reported by `.mode`, and the copy path of `Boxes3D.transform_boxes` resets it to `'xyzxyz_plus'` where `Boxes.transform_boxes` keeps it (it clones), so the only observable effect of the 3D label today is `.mode`.

Reproduced 2026-09-05 at `1eea5835`, PyTorch 2.14.0.

## Reproduction

```python
import torch
from kornia.geometry.boxes import Boxes, Boxes3D

xyzwhd = torch.tensor([[1., 2., 3., 4., 3., 5.]])
b3 = Boxes3D.from_tensor(xyzwhd, mode="xyzwhd")
print("3D mode", b3.mode, "to_tensor()", b3.to_tensor().tolist())
xywh = torch.tensor([[1., 2., 4., 3.]])
b2 = Boxes.from_tensor(xywh, mode="xywh")
print("2D mode", b2.mode, "to_tensor()", b2.to_tensor().tolist())
print("3D copy label", b3.transform_boxes(torch.eye(4)).mode, "in-place label", b3.transform_boxes_(torch.eye(4)).mode)
```

## Observed

```text
3D mode xyzwhd to_tensor() [[1.0, 2.0, 3.0, 5.0, 5.0, 8.0]]
2D mode xywh to_tensor() [[1.0, 2.0, 4.0, 3.0]]
3D copy label xyzxyz_plus in-place label xyzwhd
```

## Expected

The same default in both classes: `to_tensor(mode=None)` exporting in the stored mode. Changing the 3D default moves every `Boxes3D.from_tensor(x, "xyzwhd").to_tensor()` caller from `xyzxyz` to `xyzwhd` output, so it is a window item; batch 4c documents the split on the `Boxes3D` anchor and `to_tensor`, pins it (`test_wart_to_tensor_default_mode_ignores_the_stored_label_4251`), and carries a strict xfail for the intended behavior.

## Classification

Semantic contract / 2D-3D split (same family as #4013 and #4018). Idiom 1 (a documented label that one of two twins silently ignores).

## Disposition

**MERGE IN WINDOW — behavior change.** Canonical: `Boxes.to_tensor`'s stored-mode default. Shape: align contract, both names kept (`Boxes3D.to_tensor(mode: Optional[str] = None)`); the copy path of `transform_boxes` keeps the label like its 2D twin. Divergence today (executed): `xyzwhd` in, `xyzxyz` out for 3D; `xywh` in, `xywh` out for 2D. This is the behavior-change discussion CONTRIBUTING asks for, and the decision is: **in the coordinated Repair & Deprecation window, not before. A PR that aliases or consolidates these ahead of it will be closed and pointed here.**

Found during the bbox/boxes conventions audit batch 4c (#4020); the audit changes no runtime behavior.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
